Core/src/LongRunning/LROTraitV2.php

```diff
@@ -0,0 +1,163 @@
+        Serializer $serializer,
+        string $clientClass,
+        array $lroCallables,
+        array $lroResponseMappers,
+        $resource = null
+    ) {
+        $this->requestHandler = $requestHandler;
+        $this->serializer = $serializer;
+        $this->clientClass = $clientClass;
+        $this->lroCallables = $lroCallables;
+        $this->lroResponseMappers = $lroResponseMappers;
+        $this->lroResource = $resource;
+    }
+
+    /**
+     * Resume a Long Running Operation
+     *
+     * @param string $operationName The Long Running Operation name.
+     * @param array $lroResponseMappers A list of mappers for deserializing operation results.
+     * @param array $info [optional] The operation data.
+     * @return LongRunningOperationManger
+     */
+    public function resumeOperation($operationName, array $info = [])
+    {
+        return new LongRunningOperationManager(
+            $this->requestHandler,
+            $this->serializer,
+            $this->lroCallables,
+            $this->lroResponseMappers,
+            $this->clientClass,
+            $operationName,
+            $info
+        );
+    }
+
+    /**
+     * List long running operations.
+     *
+     * @param array $options [optional] {
+     *     Configuration Options.
+     *
+     *     @type string $name The name of the operation collection.
+     *     @type string $filter The standard list filter.
+     *     @type int $pageSize Maximum number of results to return per
+     *           request.
+     *     @type int $resultLimit Limit the number of results returned in total.
+     *           **Defaults to** `0` (return all results).
+     *     @type string $pageToken A previously-returned page token used to
+     *           resume the loading of results from a specific point.
+     * }
+     * @return ItemIterator<LongRunningOperation>
+     */
+    public function longRunningOperations(array $options = [])
+    {
+        if (is_null($this->lroResource)) {
+            throw new \BadMethodCallException('This service does not support listing operations.');
+        }
+        list($data, $optionalArgs) = $this->splitOptionalArgs($options);
+        $resultLimit = $this->pluck('resultLimit', $data, false) ?: 0;
+        $data['name'] = $this->lroResource .'/operations';
+
+        $request = $this->serializer->decodeMessage(new ListOperationsRequest(), $data);
+
+        return new ItemIterator(
+            new PageIterator(
+                function (array $operation) {
+                    return $this->resumeOperation($operation['name'], $operation);
+                },
+                function ($callOptions) use ($optionalArgs, $request) {
+                    if (isset($callOptions['pageToken'])) {
+                        $request->setPageToken($callOptions['pageToken']);
+                    }
+
+                    return $this->requestHandler->sendRequest(
+                        OperationsGrpcClient::class,
```

Comment:
Here I am using `OperationsGrpcClient` instead of `OperationsClient` of the [`DatabaseAdminClient`](https://github.com/googleapis/google-cloud-php/blob/8d3c0fd2e3f9676941d9f40f3f51aa2622d05428/Spanner/src/Admin/Database/V1/Client/DatabaseAdminClient.php#L382).

If `DatabaseAdminClient` would have used the V2 `OperationsGrpcClient`, I could have simply extracted this from the `DatabaseAdminClient->getOperationsClient` method.

Reply:
The new Operations Client will be released this Friday in https://github.com/googleapis/google-cloud-php/pull/7212 as `google/longrunning:0.4.0`